[Feat/#1,5,6] 도커 환경 지원 및 H2 DB 파일 기반으로 전환

Starlight258 opened the pull request:
### 🔗 Linked Issue
- [x] #1 
- [x] #5 
- [x] #6 

- resolved: #1    
- resolved: #6  

### 🛠 개발 기능
- `H2` DB 의존성 추가, H2 console 사용을 위해 `iframe 활성화`
- `Dockerfile`과 `docker-compose.yml`를 구성하여 **애플리케이션 컨테이너화(도커라이징)** 기반 마련
- h2 db 설정을 in-memory에서 `file 기반`으로 전환

### 🧩 해결 방법
- h2 db 의존성을 추가하여 테스트 환경 뿐 아니라 `로컬 환경에서도 작동`될 수 있도록 했습니다.
- h2-console을 지원하기 위해 `spring security에서 iframe을 허용`하도록 변경했습니다.
- dockerfile과 docker-compose 구성에 대한 설명은 길어서 정리해둔 자료로 공유해드렸습니다! :) 
 (궁금하신 부분은 댓글이나 dm 주세요! 🙌)
- h2 db의 url을 `파일 기반`으로 설정하고 `환경변수`를 사용해서 `로컬 환경과 docker 환경 모두에서 데이터가 계속 유지`되도록 했습니다.
  블로그에 관련해서 해결 방법을 작성해봤어요! 😺
  [도커 volume 설정 + 환경 변수를 사용하여 H2 DB Url을 동적으로 설정](https://shout-to-my-mae.tistory.com/395)

### 🔍 리뷰 포인트
- M1에서는 잘 작동하였는데, Intel 아키텍처에서 도커가 잘 작동하는지, db 데이터는 유지되는지 확인해주세요! 
(이미지 platform은 linux/amd64로 설정했는데, m1에서도 경고창만 뜰 뿐 잘 작동하더라구요 !)

<br>

---
### 📋 Code Review Priority Guideline
- 🚨 **P1: Request Change**
  - **필수 반영**: 꼭 반영해주시고, 적극적으로 고려해주세요 (수용 혹은 토론).
- 💬 **P2: Comment**
  - **권장 반영**: 웬만하면 반영해주세요.
- 👍 **P3: Approve**
  - **선택 반영**: 반영해도 좋고 넘어가도 좋습니다. 그냥 사소한 의견입니다.

Starlight258 committed:
[#1] feat: Add H2 database dependency and enable iframe for H2 console

Starlight258 committed:
[#5] feat: Implement Docker support and switch H2 DB to file-based storage

- Created Dockerfile and docker-compose.yml for containerized deployment.
- Changed H2 DB configuration from in-memory to file-based for persistent data storate in Docker environment.

Starlight258 committed:
[#6] chore: Add git commit message template

FacerAin commented on .gitignore:
```diff
@@ -19,3 +20,5 @@ out/
 
 
+/data/
+/data/
```
👍 P3: .gitignore에 data 디렉토리가 중복되어 기재되어 있는 것 같아요! 한번 확인 부탁드릴게요 ☺️

FacerAin replied:
👍P3: [.DS_Store](https://github.com/dnd-side-project/dnd-10th-2-backend/pull/8/files#diff-c426beb14886224177ebe8240bcbdc4b304e37b31a77ff1b4d85134bca0ea00e) 파일도 git add 되었는데, gitignore에 추가 부탁드릴게요! ☺️

Starlight258 replied:
.gitignore에 반영하고 .DS_Store은 캐시 삭제해서 git tracking 되지 않도록 수정했습니다!
추가로 전에 말씀해주셨던 [gitignore.op](https://www.toptal.com/developers/gitignore) 사이트 참고해서 내용 추가했어요 :)

FacerAin commented on Dockerfile:
```diff
@@ -0,0 +1,13 @@
+ARG JAR_PATH=./build/libs
+
+# 빌드한 파일 옮기기
+COPY ${JAR_PATH}/modutimer-0.0.1-SNAPSHOT.jar ${JAR_PATH}/modutimer-0.0.1-SNAPSHOT.jar
```
🚨 P1: Dockerfile에 버전이 직접적으로 명시가 되어있으면, 버전이 바뀔 때마다 사용자가 직접 버전 정보를 수정해야하는 불편함이 있을 것 같아요...!

따라서 `*` 기호 등을 활용하여 버전 정보를 직접 명시하지 않고 Dockerfile을 작성하는 방법이 있을 수 있어요!

관련하여 스프링 부트 공식 문서를 첨부드릴게요 ☺️
- https://spring.io/guides/topicals/spring-boot-docker/

Starlight258 replied:
그렇겠네요 !! 공식 문서 참고해서 와일드카드(`*`)로 jar 파일 실행되도록 수정했어요 !🌟

FacerAin reviewed (changes requested):
늦은 시간까지 정말 고생 많으셨어요 👍
Docker 세팅과 더불어 데이터베이스 데이터 볼륨 보장까지 꼼꼼히 신경쓰신 부분이 인상 깊어요 😄
저도 윈도우 환경에서 정상적으로 동작하는지 확인해볼게요! 감사합니다 :)

Starlight258 committed:
[#8] chore: Exclude .DS_Store from Git tracking

Starlight258 committed:
[#8] refactor: Use wildcard for JAR file in Dockerfile

- Modified Dockerfile to use wildcard(*) for JAR file selection
- Set JAR_PATH as an environment variable for use in CMD instruction
